src/main/java/me/xmrvizzy/skyblocker/mixin/HandledScreenMixin.java

```diff
@@ -80,6 +96,92 @@ protected HandledScreenMixin(Text title) {
         return skyblocker$experimentSolvers$getStack(slot, stack);
     }
 
+    @Inject(method = "drawMouseoverTooltip", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/DrawContext;drawTooltip(Lnet/minecraft/client/font/TextRenderer;Ljava/util/List;Ljava/util/Optional;II)V"), cancellable = true)
+    private void skyblocker$addTooltipComponent(DrawContext context, int x, int y, CallbackInfo ci) {
+        if (this.focusedSlot == null || this.client == null) return;
+        ItemStack stack = this.focusedSlot.getStack();
```

Comment:
Please move the implementation (the rest of this method) to a new class. Maybe `CompactorDeletorPreview` in the `item` package. You can also move `BackpackPreview` to the `item` package.

Reply:
Also, I don't know if it's worth it to implement a custom `TooltipData` like bundles which removes the need for this mixin altogether.

Reply:
Bundles have both TooltipData and TooltipComponent. The drawTooltip method HandledScreen uses converts the TooltipData using a separate method (that would add another mixin). And also the TooltipData is always slotted right underneath the Item name, you can't choose where it goes. The drawTooltip method takes the TooltipData pass it to the constructor of TooltipComponent and inserts in position 1 in the TooltipComponent list and passes that list to the drawTooltip method I am using. I think making a tooltip data would be more work.